docker-compose.yml

```diff
@@ -2,20 +2,93 @@
 # https://support.circleci.com/hc/en-us/articles/360007324514-How-can-I-use-Docker-volume-mounting-on-CircleCI-
 # For this reason, we are creating specific dockerfiles, and building the images from them, copying the files in the images
 # instead of mounting them.
+
+
 version: "3.3"
 services:
-  stardog:
+  zoo1:
+      image: zookeeper:${ZOO_TAG}
+      container_name: zkn1
+      environment:
+          ZOO_MY_ID: 1
+          ZOO_SERVERS: server.1=0.0.0.0:2888:3888 server.2=zoo2:2888:3888 server.3=zoo3:2888:3888
+
+  zoo2:
+      image: zookeeper:${ZOO_TAG}
+      container_name: zkn2
+      environment:
+          ZOO_MY_ID: 2
+          ZOO_SERVERS: server.1=zoo1:2888:3888 server.2=0.0.0.0:2888:3888 server.3=zoo3:2888:3888
+
+  zoo3:
+      image: zookeeper:${ZOO_TAG}
+      container_name: zkn3
+      environment:
+          ZOO_MY_ID: 3
+          ZOO_SERVERS: server.1=zoo1:2888:3888 server.2=zoo2:2888:3888 server.3=0.0.0.0:2888:3888
+
+  sd1:
     build:
       context: dockerfiles/
       dockerfile: dockerfile-stardog
-    container_name: pystardog_stardog
+      args:
+          - TAG=${STARDOG_TAG}
+          - NODE_TYPE=node
+    container_name: pystardog_stardog_1
+    depends_on: # This will not wait for container finish, only start (https://docs.docker.com/compose/startup-order/)
+      - zoo1
+      - zoo2
+      - zoo3
+
+  sd2:
```

Comment:
Also with multiple Stardog servers, I don't see the point in testing that clusters are 'highly' available. If you really need more than 1 node, I'd suggest running only 2 nodes, but never 3 for these tests.